### PR TITLE
ovh/cds#5005 Fix gitlab user auth with custom endpoint

### PR DESCRIPTION
**Pull request by louis-gounot**
Signed-off-by: Louis GOUNOT <louis-gounot@users.noreply.github.com>

1. Description
Force gitlab golang Client to use custom url from configuration when fecthing user info for User Authentication
1. Related issues
ovh/cds#5005
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/api/authentication/gitlab/gitlab.go [hunk 1]
```diff
@@ -100,6 +100,10 @@ func (d authDriver) GetUserInfo(ctx context.Context, req sdk.AuthConsumerSigninR
 	}
 
 	c := gitlab.NewOAuthClient(http.DefaultClient, t.AccessToken)
+	err = c.SetBaseURL(d.url)
```
Comment (richardlt) [inline engine/api/authentication/gitlab/gitlab.go]: one line error check

Comment (louis-gounot) [reply engine/api/authentication/gitlab/gitlab.go]: Hello,
Newbie to golang dev.
Do you prefer ?
```golang
	if err := c.SetBaseURL(d.url); err != nil {
		return info, sdk.WrapError(err, "invalid gitlab url")
	}
```

Comment (louis-gounot) [reply engine/api/authentication/gitlab/gitlab.go]: Done

Comment (louis-gounot) [reply engine/api/authentication/gitlab/gitlab.go]: Done

**Pull request merged**